### DISCUSSION TITLE
How access .env in extensions modules

### DISCUSSION THREAD

**Discussion by Eliophot** [directus/directus#17886] [Archive]
Hello,

How access env variables in extensions > modules ?
I try
```
import.meta.env.VITE_MYVAR
import.meta.env.MYVAR
process.env.MYVAR
```
Without success...

Thanks

**Comment by xu4wang** [comment]
Please find below an example.. no need to import anything.

```js
export default defineEndpoint((app,{env, logger}) => {
	const token = env.WECHAT_OA_TOKEN;
	logger.info(`TOKEN for Wechat OA: ${token}`);
...
```

**Reply by Ai-Yolo** [reply]
Hello, how did you access WeChat in SSO (oauth2)? I tried to configure environment variables but it was unsuccessful. I found your reply by searching keywords. I look forward to your solution.

**Reply by xu4wang** [reply]
No. I don't have WeChat SSO configured successfully. it's on my backlog.

I made Dingtalk SSO for directus last year.

**Comment by Eliophot** [comment]
In fact I think it's normal not to have access to these variables on the front end for security reasons.
To answer the problem, I made a custom endpoint (which has access to server side variables) that I call from my custom module ;)
